lib/spack/spack/detection/common.py

```diff
@@ -112,10 +112,15 @@ def path_to_dict(search_paths):
     # Reverse order of search directories so that a lib in the first
     # entry overrides later entries
     for search_path in reversed(search_paths):
-        for lib in os.listdir(search_path):
-            lib_path = os.path.join(search_path, lib)
-            if llnl.util.filesystem.is_readable_file(lib_path):
-                path_to_lib[lib_path] = lib
+        try:
+            for lib in os.listdir(search_path):
+                lib_path = os.path.join(search_path, lib)
+                if llnl.util.filesystem.is_readable_file(lib_path):
+                    path_to_lib[lib_path] = lib
+        except Exception as e:
```

Comment:
Looking at the exception hierarchy in Python, I think `Exception` is good here, but I also can't think of anything that would likely raise in this piece of code that is not an `OSError`. I guess I don't feel strongly about it:
```suggestion
        except OSError as e:
```